include/s3select_functions.h

```diff
@@ -1188,70 +1189,232 @@ struct _fn_in : public base_function
+          st = ESCAPE;
+          } else if(*p == '_' && !(*(p + 1))) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else if(*p == '%' && !(*(p + 1))) {
+        v.push_back('.');
+        v.push_back('*');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        v.push_back('.');
+        v.push_back('*');
+        st = STAR_CHAR;
+        } else  if(!(*(p + 1))) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case STAR_CHAR:
+          if(*p == escape) {
+          st = ESCAPE;
+          } else if(*p == '%' && !(*(p + 1))) {
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        st = STAR_CHAR;
+        } else if(*p == '_' && !(*(p + 1))) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else  if(!(*(p + 1))) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case ESCAPE:
+        if(!(*(p + 1))) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case END:
+        return v;
+
+      default:
+        printf("missing state\n");
+        break;
     }
+    p++;
   }
+  return v;
+}
   
   bool operator()(bs_stmt_vec_t* args, variable* result) override
   {
     auto iter = args->begin();
-    iter++;
-    base_statement* main_expr = *iter;
-    value main_expr_val = main_expr->eval();
+    int args_size = args->size();
+
+    std::string escape_char = "\\";
+    std::string like_string;
+    value main_expr_val;
+
+    if (args_size == 2) {
+      base_statement* like_expr = *iter;
+      iter++;
+      base_statement* main_expr = *iter;
+      value like_expr_val = like_expr->eval();
+      main_expr_val = main_expr->eval();
+      like_string = like_expr_val.to_string();
+    } else if(args_size == 3) {
+      base_statement* escape_expr = *iter;
+      iter++;
+      base_statement* like_expr = *iter;
+      iter++;
+      base_statement* main_expr = *iter;
+
+      value escape_expr_val = escape_expr->eval();
+      value like_expr_val = like_expr->eval();
+      like_string = like_expr_val.to_string();
+      main_expr_val = main_expr->eval();
+      escape_char = escape_expr_val.to_string();
+    }
+
+    if (escape_char.size() != 1) {
+      throw base_s3select_exception("escape character size should be 1");
+    }
+    std::vector<char> like = transform(like_string, escape_char[0]);
+    std::string str(like.begin(), like.end());
+    std::regex compiled_regex = std::regex(str);
```

Comment:
in the case like-expression is constant string (the common case), it should compiled once, it has an impact on performance.
for that it need for different syntax rules (per each use case)